runtimes/eoapi/stac/eoapi/stac/app.py

```diff
@@ -155,3 +160,32 @@ async def viewer_page(request: Request):
         },
         media_type="text/html",
     )
+
+
+if auth_settings.openid_configuration_url:
+    oidc_auth = auth.OidcAuth(
+        # URL to the OpenID Connect discovery document (https://openid.net/specs/openid-connect-discovery-1_0.html)
+        openid_configuration_url=auth_settings.openid_configuration_url,
+        openid_configuration_internal_url=auth_settings.openid_configuration_internal_url,
+        # Optionally validate the "aud" claim in the JWT
+        allowed_jwt_audiences=auth_settings.allowed_jwt_audiences,
+        # To render scopes form on Swagger UI's login pop-up, populate with mapping of scopes to descriptions
+        oauth2_supported_scopes={},
+    )
+    restricted_prefixes_methods = {
+        "/collections": [
+            "POST",
+            "PUT",
+            "DELETE",
+            *([] if auth_settings.public_reads else ["GET"]),
+        ],
+        "/search": [] if auth_settings.public_reads else ["POST", "GET"],
+    }
```

Comment:
is there a reason these `restricted_prefix` mappings are not part of the `auth_settings` too?

Reply:
Mostly because I think it would be onerous to encode in JSON. I am honestly unsure if this should be a totally env configurable copy or more of just an example. I can be convinced otherwise to add it to the settings.